common/txmgr/confirmer.go

```diff
@@ -1004,28 +1017,33 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) han
 	}
 }
 
-// EnsureConfirmedTransactionsInLongestChain finds all confirmed txes up to the depth
+// EnsureConfirmedTransactionsInLongestChain finds all confirmed txes up to the LatestFinalizedHead
 // of the given chain and ensures that every one has a receipt with a block hash that is
 // in the given chain.
 //
 // If any of the confirmed transactions does not have a receipt in the chain, it has been
 // re-org'd out and will be rebroadcast.
-func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) EnsureConfirmedTransactionsInLongestChain(ctx context.Context, head types.Head[BLOCK_HASH]) error {
-	if head.ChainLength() < ec.chainConfig.FinalityDepth() {
-		logArgs := []interface{}{
-			"chainLength", head.ChainLength(), "finalityDepth", ec.chainConfig.FinalityDepth(),
-		}
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) EnsureConfirmedTransactionsInLongestChain(ctx context.Context, head, latestFinalizedHead types.Head[BLOCK_HASH]) error {
+	logArgs := []interface{}{
+		"chainLength", head.ChainLength(), "latestFinalizedHead number", latestFinalizedHead.BlockNumber(),
+	}
+
+	if head.BlockNumber() < latestFinalizedHead.BlockNumber() {
```

Comment:
do we want to return error here ?

Reply:
I'd keep this check, just to be safe. `head` is received via callback, while `latestFinalizedHead` via direct call. In theory on super fast chains and with short actual finality depth, we might end up with `latestFinalizedHead` that is higher than `head`

Reply:
If we ditch the logs below this though, I don't think we even really need the finalizedHead here. So the difference between the latest head and finalized head may not matter. We can discuss more in the thread though

Reply:
I agree with both of you. 

@dhaidashenko , we can keep this check, but move it to the caller of this function (in this case the `processHead` in `confirmer.go`). 

I agree with @amit-momin that we can remove the reference to `latestConfirmedHead` in `EnsureConfirmedTransactionsInLongestChain`, because it's only used for the old logging purpose

Reply:
@dhaidashenko regarding [this](https://github.com/smartcontractkit/chainlink/pull/13794#discussion_r1704497844), you mentioned fast chains with short finality depth will sometimes return `latestFinalizedHead` that is higher than `head`, shall we return error for processHead in those cases ?

Reply:
I think we can keep that as an error just in `EnsureConfirmedTransactionsInLongestChain` since it's the only place we need the calculated finality depth. I'm reluctant to error out in `processHead` since other parts of the Confirmer may not be affected by this and this would be a change from the existing behavior.